desktop/cmp/filter/FilterChooser.js

```diff
@@ -98,3 +120,60 @@ const truncatedMessage = hoistCmp.factory({
         });
     }
 });
+
+//-----------------
+// Favorites
+//------------------
+function favoritesIcon(model) {
+    if (!model.persistFavorites) return null;
+    const isFavorite = model.isFavorite(model.value);
+    return Icon.favorite({
+        prefix: isFavorite ? 'fas' : 'fal',
+        className: classNames(
+            'xh-select__indicator',
+            'xh-filter-chooser-favorite-icon',
+            isFavorite ? 'xh-filter-chooser-favorite-icon--active' : null
+        ),
+        onClick: (e) => {
+            model.openFavoritesMenu();
+            e.stopPropagation();
+        }
+    });
+}
+
+const favoritesMenu = hoistCmp.factory({
+    render({model}) {
+        const options = model.favoritesOptions?.map(opt => favoriteMenuItem({...opt})),
+            isFavorite = model.isFavorite(model.value);
+
+        return menu({
+            items: [
+                menuItem({
```

Comment:
this could be replaced by the current one, floated up to the top, if it was already a favorite -- that would be very natural, I think, and allow you to easily click the x to remove it.  

Add could also be formatted by showing the favorite as it is formatted below, with just a green "add" icon on left instead of the remove cross.  Could be nice and clean